fairmq/shmem/Socket.h

```diff
@@ -125,25 +116,37 @@ class Socket final : public fair::mq::Socket
         return zmq::Connect(fSocket, address, fId);
     }
 
-    int64_t Send(MessagePtr& msg, int timeout = -1) override
+  private:
+    __FAIRMQ_ALWAYS_INLINE auto MakeMetaMsg(shmem::Message const& shmMsg) noexcept
     {
-        int flags = 0;
-        if (timeout == 0) {
-            flags = ZMQ_DONTWAIT;
+        auto const size = std::max(fMetadataMsgSize, sizeof(MetaHeader));
+
+        auto buffer = std::malloc(size);   // owning raw pointer, NOLINT
+        if (!buffer) {
+            LOG(error) << "Could not allocate shmem meta msg buffer, terminating ...";
+            std::terminate();
         }
```

Comment:
@rbx Do we want to handle this more gracefully? Returning a Send failure to the user perhaps?

Reply:
You could return `TransferCode::error`, but if malloc fails idk if it will help anything.

Reply:
> but if malloc fails idk if it will help anything.

Was thinking the same, I guess it's fine to terminate here.